components/brave_wallet/browser/simulation_response_parser.cc

```diff
@@ -0,0 +1,492 @@
+      simulation_responses::EVMSimulationResponse::FromValue(
+          json_value.GetDict());
+  if (!simulation_response_value) {
+    return nullptr;
+  }
+
+  auto simulation_response = mojom::EVMSimulationResponse::New();
+  simulation_response->action = simulation_response_value->action;
+  simulation_response->warnings =
+      ParseWarnings(simulation_response_value->warnings);
+
+  auto simulation_results = mojom::EVMSimulationResults::New();
+
+  // Parse nullable field "error" of type EVMError.
+  if (!simulation_response_value->simulation_results.error.is_none()) {
+    const auto& error_value = simulation_responses::EVMError::FromValue(
+        simulation_response_value->simulation_results.error.GetDict());
+
+    simulation_results->error = mojom::BlowfishEVMError::New(
+        error_value->kind, error_value->human_readable_error);
+  }
+
+  for (const auto& state_change_value :
+       simulation_response_value->simulation_results.expected_state_changes) {
+    auto state_change = mojom::BlowfishEVMStateChange::New();
+    state_change->human_readable_diff = state_change_value.human_readable_diff;
+
+    if (auto raw_info = ParseStateChangeRawInfo(state_change_value.raw_info)) {
+      state_change->raw_info = std::move(raw_info);
+    } else {
+      return nullptr;
+    }
+
+    simulation_results->expected_state_changes.push_back(
+        std::move(state_change));
+  }
+
+  simulation_response->simulation_results = std::move(simulation_results);
+
+  return simulation_response;
+}
+}  // namespace evm
+
+namespace solana {
+mojom::BlowfishPricePtr ParsePrice(const base::Value& value) {
+  if (!value.is_none()) {
+    const auto& price_value =
+        simulation_responses::SolanaPrice::FromValue(value.GetDict());
+    return mojom::BlowfishPrice::New(price_value->source,
+                                     price_value->last_updated_at,
+                                     price_value->dollar_value_per_token);
+  }
+
+  return nullptr;
+}
+
+mojom::BlowfishSolanaDiffPtr ParseDiff(
+    const simulation_responses::SolanaDiff& value) {
+  auto diff = mojom::BlowfishSolanaDiff::New();
+  diff->sign = value.sign;
+  if (!base::StringToUint64(value.digits, &diff->digits)) {
+    return nullptr;
+  }
+
+  return diff;
+}
+
+mojom::BlowfishSolanaStakeAuthoritiesPtr ParseStakeAuthorities(
+    const simulation_responses::SolanaStakeAuthorities& value) {
+  auto authorities = mojom::BlowfishSolanaStakeAuthorities::New();
+  authorities->staker = value.staker;
+  authorities->withdrawer = value.withdrawer;
+  return authorities;
+}
+
+mojom::BlowfishSolanaStateChangeRawInfoPtr ParseStateChangeRawInfo(
+    const simulation_responses::SolanaStateChangeRawInfo& value) {
+  auto raw_info = mojom::BlowfishSolanaStateChangeRawInfo::New();
+  raw_info->kind = value.kind;
+
+  if (value.kind == "SOL_TRANSFER") {
+    auto data_value =
+        simulation_responses::SOLTransferData::FromValue(value.data.GetDict());
```

Comment:
I think result `value.data.GetIfDict()` should be captured, checked and used in this function

Reply:
c37ccc62a37e718ade35982b6c5b8a90e3b2c8f9 - fixed + tests added